dvc/state.py

```diff
@@ -54,6 +54,7 @@ def init(root_dir, dvc_dir):
         return State(root_dir, dvc_dir)
 
     def update(self, path, md5, mtime):
+        mtime = int(mtime)
```

Comment:
This brakes comparisons in stage.py. Looks like you only need this for fsck, why not just convert it there? Though, i'm not that sure that dropping fractions of a second in mtime is a good idea.